cylc/uiserver/resolvers.py

```diff
@@ -81,6 +103,104 @@ def check_cylc_version(version):
+
+    Returns: An elaborated command.
+
+    Examples:
+        It adds one arg to a command:
+            >>> _build_cmd(['foo', 'bar'], {'set_baz': 'qux'})
+            ['foo', 'bar', '--set-baz', 'qux']
+
+        It adds one integer arg to a command:
+            >>> _build_cmd(['foo', 'bar'], {'set_baz': 42})
+            ['foo', 'bar', '--set-baz', '42']
+
+        It adds a list of the same arg to a command:
+            >>> _build_cmd(['foo', 'bar'], {'set_baz': ['qux', 'quiz']})
+            ['foo', 'bar', '--set-baz', 'qux', '--set-baz', 'quiz']
+
+        It doesn't append args == False:
+            >>> _build_cmd(['foo', 'bar'], {'set_baz': False})
+            ['foo', 'bar']
+
+        It doesn't add boolean values, just the switch
+            >>> _build_cmd(['foo', 'bar'], {'set_baz': True})
+            ['foo', 'bar', '--set-baz']
+    """
+    for key, value in args.items():
+        if value is False:
+            # don't add binary flags
+            continue
+        key = snake_to_kebab(key)
+        if not isinstance(value, list):
+            if isinstance(value, int) and not isinstance(value, bool):
+                # Any integer items need converting to strings:
+                value = str(value)
+            value = [value]
+        for item in value:
+            cmd.append(key)
+            if item is not True:
+                # don't provide values for binary flags
+                cmd.append(item)
+    return cmd
+
+
+def _schema_opts_to_api_opts(
+    schema_opts: Dict, schema: str
+) -> SimpleNamespace:
+    """Convert Schema opts to api Opts
+
+    Contains data SCHEMA_TO_API:
+        A mapping of schema options to functions in the form:
+        def func(schema_key, schema_value):
+            return (option_key, option _value)
+
+    Args:
+        schema_opts: Opts as described by the schema.
+        schema: Name of schema for conversion - used to select
+            converter functions from SCHEMA_TO_API.
+
+    Returns:
+        Namespace for use as options.
+    """
+    converters = OPT_CONVERTERS[schema]
+    api_opts = {}
+    for opt, value in schema_opts.items():
+        # All valid options should be in SCHEMA_TO_API:
+        if opt not in converters:
+            raise InvalidSchemaOptionError(
+                f'{opt} is not a valid option for Cylc Clean'
+            )
+
+        # If converter is callable, call it on opt, value,
+        # else just copy them verbatim to api_opts
+        converter = converters[opt]
+        if callable(converter):
+            api_opt_name, api_opt_value = converter(opt, value)
+            api_opts[api_opt_name] = api_opt_value
+        else:
+            api_opts[opt] = value
+
+    return SimpleNamespace(**api_opts)
+
+
+def _clean(tokens, opts):
+    """Run Cylc Clean using `cylc.flow.workflow_files` api.
+    """
+    try:
+        init_clean(tokens.pop('workflow'), opts)
+    except ServiceFileError:
+        return WORKFLOW_RUNNING_MSG
```

Comment:
There are other ways a `ServiceFileError` could be raised
```suggestion
    except ServiceFileError as exc:
        return str(exc)
```

Reply:
The `ServiceFileError` messages raised when clean fails should be pretty explanatory. (And they are not necessarily due to a running workflow e.g. https://github.com/cylc/cylc-flow/blob/c9d48a306ecac9479f3fc7a0404411f9ebece821/cylc/flow/workflow_files.py#L791-L794)

Only thing is the message for the running workflow is quite lengthy so may want to do instead
```suggestion
    except ServiceFileError as exc:
        return str(exc).split('\n')[0]
```